File: LegacyExamples/Nearables/TemperatureExample/TemperatureExample/MasterViewController.m
//
//  MasterViewController.m
//  TemperatureExample
//
//  Created by Grzegorz Krukiewicz-Gacek on 24.12.2014.
//  Copyright (c) 2014 Estimote Inc. All rights reserved.
//

#import "MasterViewController.h"
#import "DetailViewController.h"
#import <EstimoteSDK/EstimoteSDK.h>

@interface MasterViewController () <ESTNearableManagerDelegate>

@property (nonatomic, strong) ESTNearableManager *nearableManager;
@property (nonatomic, strong) NSArray *nearables;

@end

@implementation MasterViewController

- (void)viewDidLoad
{
    [super viewDidLoad];

    self.nearableManager = [ESTNearableManager new];
    self.nearableManager.delegate = self;
    
    [self.nearableManager startRangingForType:ESTNearableTypeAll];
}

#pragma mark - Segues

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
{
    if ([[segue identifier] isEqualToString:@"showDetail"])
    {
        NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
        ESTNearable *selectedNearable = self.nearables[indexPath.row];
        [[segue destinationViewController] setNearable:selectedNearable];
    }
}

#pragma mark - Table View

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
{
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return self.nearables.count;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"Cell" forIndexPath:indexPath];

    ESTNearable *nearable = self.nearables[indexPath.row];
   
    cell.textLabel.text         = [ESTNearableDefinitions nameForType:nearable.type];
    cell.detailTextLabel.text   = nearable.identifier;
    return cell;
}

#pragma mark - ESTNearableManaer delegae

- (void)nearableManager:(ESTNearableManager *)manager didRangeNearables:(NSArray *)nearables withType:(ESTNearableType)type
{
    self.nearables = nearables;
    [self.tableView reloadData];
}

@end

File: Examples/nearables/TemperatureExample-Swift/TemperatureExample-Swift/MasterViewController.swift
//
//  MasterViewController.swift
//  TemperatureExample-Swift
//
//  Created by Grzegorz Krukiewicz-Gacek on 24.12.2014.
//  Copyright (c) 2014 Estimote Inc. All rights reserved.
//

import UIKit

class MasterViewController: UITableViewController, ESTNearableManagerDelegate {

    var nearables:Array<ESTNearable>!
    var nearableManager:ESTNearableManager!
    
    override func viewDidLoad() {
        super.viewDidLoad()

        nearables = []
        nearableManager = ESTNearableManager()
        nearableManager.delegate = self
        nearableManager .startRangingForType(ESTNearableType.All)
    }

    // MARK: - Segues

    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        if segue.identifier == "showDetail" {
            if let indexPath = self.tableView.indexPathForSelectedRow() {
                let selectedNearable = nearables[indexPath.row] as ESTNearable
            (segue.destinationViewController as DetailViewController).nearable = selectedNearable
            }
        }
    }

    // MARK: - Table View

    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 1
    }

    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return nearables.count
    }

    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCellWithIdentifier("Cell", forIndexPath: indexPath) as UITableViewCell

        let nearable = nearables[indexPath.row] as ESTNearable
        cell.textLabel!.text = ESTNearableDefinitions.nameForType(nearable.type)
        cell.detailTextLabel!.text = nearable.identifier
        return cell
    }
    
    // MARK: - ESTNearableManager delegate
    
    func nearableManager(manager: ESTNearableManager!, didRangeNearables nearables: [AnyObject]!, withType type: ESTNearableType) {
        
        self.nearables = nearables as Array<ESTNearable>
        tableView.reloadData()
    }
}

